### PR TITLE
fix(users/organizations/auth): prevent 500 on signin when org deleted without cascade (#3709)

### LINKED ISSUE #3709: Signin 500 (null._id) when a user's active membership points to a deleted org — removeById skips cascade
## Symptom

Sign-in returns **HTTP 500** with `Cannot read properties of null (reading '_id')` for any user who has an **ACTIVE membership whose organization no longer exists**. The error is unhandled (`signin` has no try/catch) and bubbles to Express.

Observed live on `TrawlNodeDev` 2026-05-27: a user (`test@gmail.com`) had one active membership + `currentOrganization` pointing at an org that had been deleted; every sign-in 500'd. Clearing the dangling membership + nulling `currentOrganization` fixed it immediately.

## Root cause (two layers)

**1. Non-cascading org deletion leaves dangling refs.**
`modules/users/services/users.service.js:126` deletes a user's organizations via `OrganizationsCrudService.removeById(orgId)`:
```js
// organizations.crud.service.js:299
const removeById = (id) => OrganizationsRepository.remove({ _id: id });   // bare delete, NO cascade
```
Unlike `remove(organization)` (same file, ~L177) which **does** cascade (deletes memberships, reassigns/clears every affected user's `currentOrganization`), `removeById` just drops the org document. So when a user is deleted / closes their account, any **co-members** of that user's orgs are left with active memberships + `currentOrganization` pointing at a now-missing org.

**2. Sign-in null-derefs on the dangling ref instead of tolerating it.**
`modules/organizations/services/organizations.crud.service.js:262` (`autoSetCurrentOrganization`, called first thing in `signin`):
```js
const orgId = memberships[0].organizationId._id || memberships[0].organizationId;
```
When the membership's `organizationId` is populated and the org was deleted, the populated value is `null` -> `null._id` throws. The signin path (`auth.controller.js`) assumes a populated org is always non-null.

## Fix

- **Root cause**: route the user-deletion org cleanup through the cascading `remove(organization)` (or give `removeById` the same membership + `currentOrganization` cascade). Callers that drop an org must never leave orphan memberships.
- **Defensive (belt-and-suspenders)**: in `autoSetCurrentOrganization` (and any signin populate path), treat a membership whose populated `organizationId` is null as invalid — skip / clean it — instead of dereferencing. Pick the first membership with a *live* org; if none, set `currentOrganization: null`.
- Wrap `signin` org-resolution so a data-integrity glitch degrades to "no current org" rather than a 500.

## Repro

1. User A and User B both members of Org X (B's membership active, `currentOrganization = X`).
2. Delete User A (whose deletion removes Org X via `users.service.js:126` -> `removeById`, no cascade).
3. User B signs in -> 500 `Cannot read properties of null (reading '_id')`.

## Notes

- Devkit-core (auth + organizations modules) -> fix upstream here, then propagate to downstreams via `/update-stack`.
- Found while shipping an unrelated Trawl Vue UI PR (comes-io/trawl_vue#935); filed as orphan tracking issue (no assignee).

### PR DESCRIPTION
**Pull request by PierreBrisorgueil**
## Summary

Fixes #3709 — `Cannot read properties of null (reading '_id')` 500 on signin when a user has an active membership pointing at a deleted organization.

### Root cause

When a sole-owner user is deleted, `users.service.js:remove()` called `OrganizationsCrudService.removeById(orgId)` (bare DB delete, no cascade). This left co-members with `currentOrganization` pointing at the now-deleted org. On the co-member's next signin, `autoSetCurrentOrganization` called `MembershipRepository.list()` which populates `organizationId` — Mongoose returns `null` for deleted orgs — then crashed on `null._id`.

### Three-layer fix

**Layer 1 (`users.service.js`):** Replace the bare `removeById` call with an inline cascade that:
- Collects all co-members whose `currentOrganization = orgId` (affected users)
- Deletes all memberships for the org
- For each affected co-member, finds their next available org and updates `currentOrganization` (or sets null if none)
- Deletes the org via bare `OrganizationsRepository.remove` (preserving task data, consistent with previous behavior)

**Layer 2 (`organizations.crud.service.js`):** Harden `autoSetCurrentOrganization`:
- Filter memberships where `organizationId` is `null` after Mongoose populate before dereferencing `._id`
- Guard the early-return branch: only trust a membership as still-active when its populated `organizationId` is non-null

**Layer 3 (`auth.e2e.tests.js`):** Add regression test verifying signin returns 200 (not 500) when `currentOrganization` is a bogus ObjectId (pre-existing data-integrity anomaly scenario, belt-and-suspenders).

### Tests added

- `modules/users/tests/users.remove.cascade.e2e.tests.js` — full repro: User A (sole owner) deleted → Org X removed → User B (co-member) can sign in without 500
- `modules/organizations/tests/organizations.crud.dangling-org.unit.tests.js` — 4 unit tests covering all null-populate edge cases in `autoSetCurrentOrganization`
- `modules/auth/tests/auth.e2e.tests.js` — 1 new test: data-integrity corruption → signin still 200

### Test results

- Unit: 105 suites, 1525 tests — all pass
- Integration: 28/29 suites pass (1 pre-existing flaky timing test on master too)
- E2E: 5 suites, 11 tests — all pass

### CODE REVIEW

File: modules/users/services/users.service.js [hunk 1]
```diff
@@ -119,12 +120,20 @@ const remove = async (user) => {
...
+        // Step 2: Delete all memberships for this org (including this user's and all co-members')
         await MembershipService.deleteMany({ organizationId: orgId });
-        await OrganizationsCrudService.removeById(orgId);
-        continue; // membership already deleted above
+        // Step 3: For each affected co-member, switch to their next available org or set null
+        await Promise.all(affectedUsers.map(async (u) => {
+          const remaining = await MembershipRepository.list({ userId: u._id, status: MEMBERSHIP_STATUSES.ACTIVE });
+          const nextOrg = remaining.length > 0 ? (remaining[0].organizationId._id || remaining[0].organizationId) : null;
+          await UserRepository.updateById(u._id, { currentOrganization: nextOrg });
+        }));
```
[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply modules/users/services/users.service.js:134]: Fixed: the deletion-cascade in `users.service.remove` now applies the same `liveMemberships = remaining.filter((m) => m.organizationId != null)` guard before dereferencing `organizationId._id`, mirroring the identical fix already present in `autoSetCurrentOrganization`. Unit tests added covering both the null-only case (sets currentOrganization to null) and the mixed case (skips null membership, picks first live org).
